### ISSUE TITLE
Jenkinsfile diagnostics incorrectly says options can only be used once inside the pipeline block

### ISSUE THREAD

**Issue by kluisi** [sgwozdz/jenkinsfile-support#55]
**Describe the bug**
In VSCode, the syntax highlighter and diagnostics report that a per-stage options block is not allowed, but according to the Jenkins documentation (https://www.jenkins.io/doc/book/pipeline/syntax/#options), a per-stage options block is allowed with limited options.

**Expected behaviour**
I expect the following code to not report an error:

```
pipeline {
    agent any

    options {
        gitLabConnection('myGitlabInstance')
    }
    stages {
        stage('Print Environment'){
            options {
                warnError("Print an error")
            }
            steps {
                sh "printenv"
            }
        }
    }
}
```

**Actual behaviour**
The Print Environment stage options block is underlined in red and reports the following peek problem:

```
Required: No
Parameters: None
Allowed: Only once, inside the pipeline block.

It is not allowed to use options { more than once
Peek Problem (⌥F8)
No quick fixes available
```

**Screenshots**
![image](https://user-images.githubusercontent.com/35928432/91489417-cf51f680-e87e-11ea-8302-83318db821e9.png)
![image](https://user-images.githubusercontent.com/35928432/91489698-3b345f00-e87f-11ea-8113-9c24d892ae2c.png)

**Additional context**
None

**Comment by asherber** [comment]
FWIW, the Jenkins docs are not consistent here. https://www.jenkins.io/doc/book/pipeline/syntax/#options does say that the options directive is allowed only once, inside the pipeline block, but if you scroll down farther there's a section called 'stage options'.

**Comment by sfrank-mc** [comment]
I think you mean the "Pipeline Options" rather than the "Stage Options"?
![image](https://user-images.githubusercontent.com/67592530/108363567-fc0f1700-71ec-11eb-8c70-efbad9b20afd.png)

**Comment by asherber** [comment]
I'm not sure what you mean. At the link I posted, the page says that the options directive is only allowed once, inside the pipeline block. I'm guessing this is the rule that this extension is enforcing. 

However, if you scroll down a little farther, you'll find the text you highlighted, which says that options are also allowed inside a stage. This contradicts the statement earlier on the page, but it is correct -- you can have options inside a stage.

**Comment by sfrank-mc** [comment]
I think https://www.jenkins.io/doc/book/pipeline/syntax/#options refers to the "pipeline options" directive - which can appear once and only at the pipeline level.
While https://www.jenkins.io/doc/book/pipeline/syntax/#stage-options refers to a different yet similar directive for "stage options" which can appear at each stage.
So I don't think there's a contradiction.

**Comment by asherber** [comment]
If you look at the way the page is formatted, 'stage options' is a subset of 'options'. Anyone clicking on 'options' in the TOC will be told that options can only appear once. Only if you happen to scroll down will you be told that actually, that's not true.

If you look at [environment](https://www.jenkins.io/doc/book/pipeline/syntax/#environment), for example, the page explicitly says that this directive is allowed inside the pipeline block or within stage directives.

In any case, the important thing is that options _is_ allowed inside a stage, and this extension is incorrectly flagging that as an error. I was just pointing out that it's understandable why the extension is doing this, since the docs are not as clear as they could be.

**Comment by sfrank-mc** [comment]
I understand what you mean... I interpreted it differently from you but yours is a valid interpretation as well.
In any case like you said... sound like this could be made clearer by their docs - so maybe a documentation bug for them may help that - but it is clearly a valid use-case...

**Comment by asherber** [comment]
Yes, I opened https://github.com/jenkins-infra/jenkins.io/issues/4123
